### ISSUE TITLE
Provisional release notes for CR-6 community firmware release 5 beta

### ISSUE THREAD

**Issue by Sebazzz** [CR6Community/Marlin#84]
First of all I want to thank everyone who has been [testing the alpha 4 release of the community firmware](https://github.com/CR6Community/Marlin/releases/tag/v2.0.7.1-cr6-community-release-4-alpha). All the testing and feedback has made this firmware better and better. This is *truly* a community firmware.

In the period between then and now, my Creality CR-6 has died and from its ashes it became alive once again with a [BigTreeTech SKR CR-6](https://damsteen.nl/blog/2020/11/25/how-to-btt-skr-cr6-installation) board. A while later I also received a new Creality CR-6 SE [with the v4.5.3](https://www.reddit.com/r/CR6/comments/k97r5d/creality_cr6_se_new_motherboard_v453/) motherboard. It took a while for that one to arrive, then a while for the firmware source code to be released, so it has caused some delays to get real work done. Yet, some users were able to test some fixes I was able to do without a operational printer.

*TL;DR: CR-6 MAX support, SD card folder support, better Octoprint support, BigTreeTech SKR CR-6 support, native and improved filament sensor support, many stability and ABL accuracy improvements, rebased on latest Marlin, and enabled the watchdog safety feature*

## Table of contents

- Special thanks
- Other news
- Major features (all firmware editions)
- Detailed changes
- Comparison with Creality stock firmware
- Known issues
- Flashing instructions
- Files listing
- FAQ
- Reporting issues
- Contributing

## Special thanks to...

- @grobux for designing a completely new user interface for Creality stock touch screen
- @Thinkersbluff, @Beeble2695 and @InsanityAutomation as a sparring partner for various firmware and hardware mechanics of the CR-6
- @InsanityAutomation for further refining the bed leveling code and submitting some things upstream
- @Nushio for putting up testing protocols and doing an insane amount of stress testing on the firmware
- @dabbodev for contributing configuration changes to support for the Creality CR-6 MAX
- The Discordians in the #prerelease-testing channel for doing the initial pre-release testing
- The various testers and contributors to the firmware, either through testing, feedback, as a sounding board to the questionnaire we sent out earlier, or even donations!

If you like to contribute to the community firmware, in one way or the other, [please check the home page of the Creality CR-6 community firmware](https://github.com/CR6Community/Marlin#credits).

## Other news

- @InsanityAutomation is using the CR-6 community firmware and rebranding it for use on machines sold [by Tiny Machines](https://www.tinymachines3d.com/)
- Some auto bed leveling features have already been merged to upstream Marlin

## Major features across firmware editions

_Note: This is *still* a beta formally - because this is based on a unreleased version of Marlin. However, thanks to the amount of effort that has gone into it, I'm confident it should be a pretty stable release!_

### Safety

**Important safety feature: We now enabled the watchdog for STM32F1 processors - if Marlin would freeze from some reason (like due to SD card issues) the printer is automatically reset and the heaters are disabled - this feature is not enabled on the stock Creality or BTT firmware**

### User interface

This release features an entirely refreshed user interface by @grobux! Improved font spacing, improved graphics. JPEG artifacts be gone!

![image](https://user-images.githubusercontent.com/1426097/103158845-d0f6e100-47c2-11eb-96af-51c3fbfa4494.png)

In addition we implemented toggles allowing *you* to decide if you want to hear sounds from the touch screen (or not), and if you like the touch screen display to dim to 10% on idle (30 seconds currently).

![qol-settings](https://user-images.githubusercontent.com/1426097/103174785-1f5cbc00-4865-11eb-9c47-04ecc2a3c143.jpg)

We've also implemented that you can tune the flow rate (the amount of filament extruded) during the print from the "tuning" screen. Very useful when tuning in new filaments. You can also control and review the fan speed percentage from the tuning screen.

![flow-rate-tuning](https://user-images.githubusercontent.com/1426097/103160420-276d1b00-47d5-11eb-8020-fe05c1742c01.jpg)

You can also now quickly set the bed and hot-end temperature by tapping the temperature in the footer.

On the print screen we've added displays for the fan speed, flow rate, and X/Y/Z axis positions:

![image](https://user-images.githubusercontent.com/1426097/103288030-45f53100-49e4-11eb-8e4c-fa994bec14b5.png)

On this screen you can also set the hotend and bed temperature by directly tapping the numbers.

### SD card folder support

We support navigation through folders on your SD card. No longer are you limited to 5 pages of gcode files. You can organise your SD card in folders.

![folder-support](https://user-images.githubusercontent.com/1426097/103160423-2b993880-47d5-11eb-8a81-af986e69b511.jpg)

### Hardware support

- Support for Creality v4.5.2 and the new v4.5.3 motherboards. See the [FAQ](#faq) section to know which board you have.
- Support for [the BigTreeTech SKR CR-6 board](https://damsteen.nl/blog/2020/11/25/how-to-btt-skr-cr6-installation), both with BTT TFT and stock Creality DWIN TFT
- Support for the Creality CR-6 MAX

### Octoprint

We have improved Octoprint support. We show the printer status messages also on the home screen.

![image](https://user-images.githubusercontent.com/1426097/103352969-e95b4a00-4aa7-11eb-87ab-623bd5dbbfd9.png)

The filament sensor is working and Octoprint acts on it - it pauses the print. Make sure you have the correct gcode in Octoprint so it parks the nozzle.

![image](https://user-images.githubusercontent.com/1426097/103352935-d6e11080-4aa7-11eb-885e-187ea3f47d97.png)

In addition when an Octoprint print starts we automatically show the "printing" screen with full access to all details and tuning options. You can also set the print progress using the [the detailed progress plugin](https://plugins.octoprint.org/plugins/detailedprogress/).

### Reliability
- Rebased on latest Marlin bugfix-2.0.x sources which will eventually become Marlin v2.0.7.3
- Increase bed leveling and homing (Z-offset) accuracy
   - We do this by preheating before probing and turning off heaters when probing so electrical interference when reading the strain gauge is minimized
   - Homing and probing is perceived slower - but in terms of print time this is just a fraction of the time
   - By default we probe at 120C/50C - but if you have preheated your printer to a higher temperature, then that temperature is kept
   - Disable this behavior by commenting `PREHEAT_BEFORE_PROBING` and `PROBING_HEATERS_OFF` in the firmware configuration
- Further tinkering with the movement settings for optimum performance
- Again for Octoprint users: filament sensor is supported

##  Detailed Changes 

- Both homing and leveling preheat to decrease the effect of temperature differences and filament stuck to the nozzle causing issues
- Pausing a print will no longer pause the print with the nozzle hanging over the print, but will instead now park the nozzle during the pause
- Heaters are turned off while the actual probing is done. This reduces electrical noise which may otherwise cause inaccuracies in the Z-Offset and Bed Leveling measurements.
- Improve reliability of probing and leveling (issue #71, issue #67, issue #49, issue #42, issue #41, issue #38)
- Increase serial buffer size to improve reliability when printing a lot of vases via a host like Octoprint
- Automatically enable bed leveling after homing - even if you haven't called `M420 S1` before
- Enable `G26` bed leveling mesh validation tool
- Enable PID tuning for the heated bed
- Enable hotend time-out. This feature automatically disables the hot-end to prevent charring and jams. Currently set to 5 minutes.
- Enable G2/G3 arc by default - useful [for those using Arc Welder](https://plugins.octoprint.org/plugins/arc_welder/)
- Enable host action commands and host prompt support
- Enable probe accuracy test M48
- Report PID tuning progress on the status messages - you still need to run [M303](https://marlinfw.org/docs/gcode/M303.html) manually (through serial monitor or gcode on your SD card) but you can now view the results as it happens
- Tap the temperatures in the footer to set them

### Changes specific when using Creality v4.5.2 and v4.5.3 motherboards

- Improve reliability for SD card prints freezing - especially if using v4.5.2 boards (issue #62)
- Improve reliability in vase mode, prevent blobs when printing vases

### Changes specific when using the Creality stock display

- When printing from a host, show 'Printing from host' instead of 'DWIN technology' as title (issue #55 and #50)
- For some users M600 was not working - reliability has been improved (issue #58)  - [please read here how to use it](https://damsteen.nl/blog/2020/12/26/how-to-do-filament-change-with-cr6-community-firmware-release-5)
- When printing from a host and removing the SD card, the home screen was displayed (issue #49)
- Improve the flashing instructions with instructions for Windows, MacOS and Linux users.
- Its now possible to change the flow rate (amount of filament extruded) from the "Tune" screen during printing
- Fixed G29 in start gcode causing unexpected behaviour (issue #87)
- If you are using an all-metal hot-end you can now enter temperatures up 450 degrees on the touch screen. You still would need to recompile the firmware so this temperature is actually allowed by Marlin.

### Changes specific to BTT SKR CR6 boards

![image](https://user-images.githubusercontent.com/1426097/103158817-8e350900-47c2-11eb-956f-4bd48bdbd3d6.png)

- Tune Trinamic motor current values to optimum values:
   - Extruder current has been set down - especially important for those of us using full metal extruders to prevent filament melting
   - Z-stepper current has been increased - both stepper motors share the same stepper driver and I have observed skipped steps otherwise
- Set spreadcycle stepper profile to 24V because the steppers are driven with 24V voltage
- Enable TMC driver monitoring to prevent skipped steps in case of overheat conditions. When the stepper drivers are about to get too hot the stepper driver current is lowered to prevent shutdown. This also allows the firmware to report if the drivers overheat. 
- Improve filament sensor reliability
- Enable configuration of stealthchop/spreadcycle hybrid threshold [via M913](https://marlinfw.org/docs/gcode/M913.html). If you don't like to tune this: no worry - everything defaults to stealthchop.
- Enable motherboard fan only when the printing (= Z steppers are moving). This makes for a *very* silent printer when idle.
- Tune PID - the new defaults should work well for most users
- Use `FAN_KICKSTART_TIME` to ensure the part cooling fan is started reliably
- Enable quick homing
- Prevent cancelling a print on the first layer from scratching the build plate
- Improve sorting of SD card files

### Changes specific to BTT SKR CR6 firmware with BTT TFT

- Enable filament sensor - it didn't work if using BTT TFT
- Enable jerk editing from MarlinUI or via gcode `M205`
- Add support for bed leveling in the LCD menu
- Add a CR-6 branded bootscreen
- Change default preheat options to sensible PLA and PETG temperatures
- Enable the Z-offset wizard (available in Advanced Configuration -> Probe offsets)
- Add information menu to the main menu
- Add "mute" option to mute the speaker
- 🐍, 🛸 and 🧱 (scroll down in the main MarlinUI menu....)

## Comparison to Creality firmware

See for a full list the [Community firmware 4 alpha release notes](https://github.com/CR6Community/Marlin/releases/tag/v2.0.7.1-cr6-community-release-4-alpha), but some highlights here:

- Allow the probe offset to be set in 0.01mm offsets (issue #15)
- Pre-heat PLA and PETG temperatures are properly shown in the touch screen interface
- Save power loss recovery settings to the SD card instead of EEPROM, preventing blobs

- Filament runout detection has now been enabled by default again. (issue #19) 
   - We use the Marlin native filament runout detection. This is more reliable than the home-brew implementation of Creality because it debounces any possible false readings. Give your filament sensor a chance again.
   - If you have previously used the resistor trick to bypass the filament runout sensor, you can undo it. If your filament runout sensor is really defective, you can disable it using `M412 S0` - otherwise you can enable it using `M412 S1`.

- Support `M300` (play tone / buzzer) (issue #20) - You can now let your printer make some noise! Note that we only support the duration parameter of M300, and only with 8 millisecond precision due to the touch screen limitation.
   - This is especially useful because you can add M300 in your end gcode to have an audible notification that your print is done
   - When you have a filament change pending, the printer will notify you using an audible alert
   - Errors l

- Support `M73` (print progress) (issue [#7](https://github.com/CR6Community/CR-6-touchscreen/issues/7) and issue #27)
   - When printing from Octoprint you can use [the detailed progress plugin](https://plugins.octoprint.org/plugins/detailedprogress/) to update the progress on the printer display
- Support `M75`/`M77` (start/stop print timer) on the touch screen level. This can be useful if printing from a host, because if you issue this gcode, the touch screen changes to the print progress screen. When printing from Octoprint, this will be injected by default in my experience.

- Enable EMERGENCY_PARSER commands 
   - Enable support for gcode command M0 with resume via M118 via the touch screen (the "power loss recovery" screen is (ab)used for this)
 
- Support `M117` LCD message (issue [#7](https://github.com/CR6Community/CR-6-touchscreen/issues/7)) on the print progress screen
   - If you use Cura, you can inject "time remaining" or the "current layer" and it will show on the display

- Enable linear advance / pressure advance allowing you to do [linear advance calibration](https://teachingtechyt.github.io/calibration.html#linadv)
- Based on recent version of Marlin (as mentioned above) instead of an early version of Marlin 2.0.0

- Support `ADVANCED_PAUSE_FEATURE`

    - Support `M600` filament change  - [please read here how to use it](https://damsteen.nl/blog/2020/12/26/how-to-do-filament-change-with-cr6-community-firmware-release-5)
    - Support `M701`/`M702` filament load/unload

### Some bugs fixed

Some bugs *may* be fixed in the Creality v2.x firmware too - this list compares to the Creality v1.0.3.7 firmware on which the BTT SKR firmware is also based.

- Fix temperature reporting issues in Octoprint (issue #3)
- Don't execute the SD abort gcode sequence when a print is succesfully finished - you can add `G1 X0 Y{machine_depth} ;Present print` to your end gcode in Cura to put the bed forward after the print completes.
- Prevent first layer abort from scratching the bed (issue #12)
- See for a full list the [Community firmware 4 alpha release notes](https://github.com/CR6Community/Marlin/releases/tag/v2.0.7.1-cr6-community-release-4-alpha), but some highlights here:

## Known issues

- Auto-bed leveling might be slower than what you are used to - but this trade off is in favor of accuracy. We recommend leveling once - the printer will remember the leveling settings. Remove `G29` from your start gcode - there is no need to level every time you print. [Check this page for a recommended](https://damsteen.nl/blog/2020/10/25/creality-cr6-start-print-without-drooping) start gcode for your slicer. Alternatively if you're tech-savvy, you can undefine `PREHEAT_BEFORE_PROBING` in the Configuration.h.
- Heaters stay on after homing or leveling is completed - but not on a level that is harmful to PLA. The heaters are turned off though once you leave the leveling menu.

**When using the Creality stock touch screen:**
- M600 does a filament purge and then continues printing - it does not wait for you to confirm enough filament is purged. For this reason I recommend to do the purging manually  yourself and set the purging amount in the slicer to a low value like 10mm. This is due to a limitation in Marlin.
- The movement screen is less reliable if you have not homed before using the movement controls. Please home first before attempting to move the axes. We're showing the positioning numbers in red to remind you of that (issue #73)

**For the BTT SKR CR6 board:**
- `EMERGENCY_PARSER` support is not available due to [upstream bug #19623](https://github.com/MarlinFirmware/Marlin/issues/19623) - so the commands like `M112` will be executed in order. This issue is also present in the stock firmware.
- Homing heats up before probing and this is not shown on MarlinUI so it appears not to function if you home from within the Marlin UI menu. If you go to the home/info screen though, you can see it heat up though.

## Features not available in this firmware

- Full localization support. The Creality v2.x firmware offers full support for multiple languages - the CR-6 community team has no manpower to maintain localizations. If you're a bit tech savvy you can fork our touch screen firmware and [maintain a set of localized screens as detailed here](https://github.com/CR6Community/CR-6-touchscreen/issues/16#issuecomment-727881233). Get in touch and we'Il put a link up to your fork!

## Flashing instructions

**Note: This release is accompanied by a new firmware for the touch screen. This release _and_ the new touch screen firmware goes hand-in-hand - one cannot be used without the other.**

- It doesn't matter if you flash the touch screen on mainboard firmware first. 
- Make sure you pick the right firmware package for your particular hardware configuration!
- You can get the files in the "assets" section below.

### Touch screen flashing instructions

**This release of the touch screen firmware comes with a new version of DGUS OS (the underlying operating system of the touch screen) to fix text rendering issues.**

Flashing instructions are [also available as a video](https://www.youtube.com/watch?v=Jswzrh2_ekk) - this video is for the Ender 3 V2 but the same procedure applies to the CR-6. For instructions, check the README.txt in the touch screen firmware download.

For this release the touch screen firmware and flashing instructions is included in the firmware release package. Be sure to follow the instructions carefully. If you experience issues, try flashing the touch screen a second or even a third time. It doesn't always "take" all the files. Sorry - hardware limitation!

### Mainboard / Marlin flashing instructions

0. If you like to remember what settings you had, run `M503` in a serial monitor.
1. Use an empty SD card formatted FAT32 4096KB sector size.
2. Put the firmware ".bin" file downloadable from the 
3. When using Windows - do a "safe removal" to remove your SD card.
4. Put the SD card in the printer.
5. Either turn the printer off, then on again or issue command `M997` from Octoprint/Pronterface. 
   - BTT SKR boards will flash the hot-end led while the firmware is flashed and rename the `firmware.bin` file to `firmware.cur`.
   - Creality boards will give no indication of correct flashing - except for the fact that it took a few seconds for the flashing to complete
6. Verify the firmware is flashed by issuing `M115` or going to "Control -> Info" on the touch screen.
7. You can now remove the file from the SD card.
8. **After flashing go to the menu and reset to factory settings or issue `M502` followed by `M500` in Octoprint - do this also if you have been running an earlier release**. You need to relevel the bed, recalibrate the z-offset, reprogram esteps (if you've changed them from defaults), and do a PID tune (if you find that the heating is not reliable for you or if you use high temperature plastics like ABS and nylon).

### File listing

The files can be downloaded from the assets section at the bottom of the page. You can find the description of the files below.

```
File name                                                        Description                                                SHA256 hash
---------                                                        -----------                                                -----------
CF5Beta-btt-skr-cr6-with-btt-tft-2020-12-30-14-17.zip            CR-6 SE with BigTreeTech SKR CR6 and BTT TFT               8857E1B8B7711BFDC1A929CB72D6A39708F4D961FEEECC879EFF0C06C1C39349
CF5Beta-btt-skr-cr6-with-stock-creality-tft-2020-12-30-14-18.zip CR-6 SE with BigTreeTech SKR CR6 and stock Creality screen B1381FCC664ABF0881A8F004317372959683530D17367B859062C1B63EED17A7
CF5Beta-cr6-max-2020-12-30-14-18.zip                             CR-6 MAX                                                   53E9E79819DB3A6F3F7D748AAC558450D6BBD64BDA297F5E7A8AB8EE9A6A4368
CF5Beta-cr6-se-v4.5.2-mb-2020-12-30-14-19.zip                    CR-6 SE (v4.5.2 motherboard)                               68429FA9A3EF8991350F487C502A9E2BE17EEB6A32AA9E006AE23BBB8F778B43
CF5Beta-cr6-se-v4.5.3-mb-2020-12-30-14-19.zip                    CR-6 SE (v4.5.3 motherboard)                               FD5F324A6854780128C4C1029D24BA0C551965AC255D16E87369783AC54AAD42
```

As mentioned above, while touching the bed we turn off the hot-end momentarily. If you don't want that, you can use the builds below. _Only use these if you have a good reason to do so._

```
File name                                                                      Description                                                SHA256 hash
---------                                                                      -----------                                                -----------
CF5Beta-ProbeHeaterOn-btt-skr-cr6-with-btt-tft-2020-12-30-14-23.zip            CR-6 SE with BigTreeTech SKR CR6 and BTT TFT               8DAB7511366D728404AA2DBA0F7E71F79E8159F2C59696CFD563532921F3E15B
CF5Beta-ProbeHeaterOn-btt-skr-cr6-with-stock-creality-tft-2020-12-30-14-23.zip CR-6 SE with BigTreeTech SKR CR6 and stock Creality screen CF454E418F29B1A1DC63FE7029BDD51A357A6233AC60627CCC6AD4A0406AF4F2
CF5Beta-ProbeHeaterOn-cr6-max-2020-12-30-14-24.zip                             CR-6 MAX                                                   BAB76A8F420A1382C0467D4C404E256C54D091E40774C15C0F6AB49400103161
CF5Beta-ProbeHeaterOn-cr6-se-v4.5.2-mb-2020-12-30-14-24.zip                    CR-6 SE (v4.5.2 motherboard)                               B078D475192536F78B5EBB804C988834CB321700BAC8A669DCE28D5F426B22AC
CF5Beta-ProbeHeaterOn-cr6-se-v4.5.3-mb-2020-12-30-14-25.zip                    CR-6 SE (v4.5.3 motherboard)                               3430FA98BF96B99270DA2F789292759D5D0A14E3CF67D14EBD611699EB51CB26
```

*Note: The files with `ProbeHeaterOn` keeps the hot-end heated while probing - in our experience this decreases bed leveling accuracy and cause a drifing Z-offset - but some people prefer it this way. You have a choice.*

## Rollback instructions

To completely roll back to an earlier release:

- Flash the [v1.0.0-a1](https://github.com/CR6Community/CR-6-touchscreen/releases/tag/v1.0.0-a1) touch screen firmware and [Community release 3](https://github.com/CR6Community/Marlin/releases/tag/v2.0.7.1-cr6-community-release-3)
- Flash Creality stock firmware and touch screen using the instructions Creality provided

## FAQ

Some frequently asked questions. Your question will most likely be answered here!

### How do I know if I have a Creality v4.5.2 or v4.5.3 motherboard?

- You can open your printer - the motherboard version found in the area between the stepper motor connectors and the connectors of the flat cables [as shown here](https://www.reddit.com/r/CR6/comments/k97r5d/creality_cr6_se_new_motherboard_v453/)
- If you have a CR-6 with a serial number before December 2020 (P202J*[month]*K[number]) you will have a v4.5.2 motherboard
- If your stock firmware was v1.0.3.7 or v1.0.2 you have a v4.5.2 motherboard
- If your stock firmware was v1.4.x you have a v4.5.2 motherboard

### Does it matter if I flash the touch screen or motherboard firmware first?

No - you can do it in either order. 

### I'm having trouble flashing the touch screen firmware / things aren't working / I'm seeing garbled or missing text

Try reflashing your touch screen firmware. Read [the flashing instructions](https://github.com/CR6Community/CR-6-touchscreen/blob/extui/src/README.md) *carefully*. The DWIN stock touch screen is *very* picky when it comes to SD cards. Sorry, we nor Creality can't change this.

### After flashing the touch screen it only shows a black display

This has occurred to some users while upgrading to either official or community firmware, but it is unclear why this happens. 

Make sure you haven't accidentally shorted the PCB of the touch screen when you attempted to flash it. Because the DWIN touch screen uses SRAM, it takes a while for the error condition to clear. Turn off the printer for 5 minutes or more and try reflashing the touch screen.

### My stepper motors run the wrong way (BTT SKR CR6 board only)

There has been one report for the BTT SKR CR6 board by which the stepper motor turns the wrong way. You need to compile the firmware yourself. For most users the direction is correct though.

## Reporting issues

Please [follow the instructions here carefully](https://github.com/CR6Community/Marlin/blob/extui/.github/issue_template.md).

## Troubleshooting

- Verify that the issues are not in your hardware. We've seen issues that were reported as firmware issues, but were actually hardware issues that "just" happened.
    - Check you have a good SD card. Do **not** use the Creality SD card that came with your printer. It will fail sooner or later and result in failures. Try a different SD card.
    - Leveling: try it again a second time and try using the stock cable clips.

- Verify that the issues are not in your slicer
   - Cura 4.7.x is having issues
   - Cura 4.8 is still giving some issues for some people. Try playing with the maximum resolution and maximum deviation settings in your slicing profile.
   - 
## Feature requests

Feature requests are welcome! Let us know in [the issue tracker](https://github.com/CR6Community/Marlin/issues/)!

## Contibuting

Your contributions are very welcome! 

- Please open [an issue in the issue tracker first](https://github.com/CR6Community/Marlin/issues/) to discuss what you like to contribute. This is especially important if it involves the touch screen because due to technical limitations only one person can work on the touch screen firmware at the same time.
- Issues marked "[good first issue](https://github.com/CR6Community/Marlin/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)" or "[help wanted](https://github.com/CR6Community/Marlin/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)" can are a good issue to start with!

## Some final notes

- Remember that this is a community project, we're not paid (definitely not by Creality!), we have a full time job, doing this in our spare time, and we're just 3d printing enthusiasts who like to have a better Creality CR-6 experience for everyone.
- We're active in the [CR-6 community discord](https://discord.gg/RKrxYy3Q9N) - come say hello!
- You can get the files in the "assets" section below. Note that if you like to compile the firmware yourself [you can best pick the correct branch from Github instead](https://github.com/CR6Community/Marlin/#development-and-compile-it-yourself)
- If you like to contribute to the community firmware, in one way or the other, [please check the home page of the Creality CR-6 community firmware](https://github.com/CR6Community/Marlin#credits).

The CR-6 community firmware team wishes you a happy and healthy 2021!

__Downloads below - find the correct download in the file listing above__

**Comment by Thinkersbluff** [comment]
“ Enable G29 bed leveling mesh validation tool” may be typo?  
Mesh validation tool is G26.

**Comment by Sebazzz** [comment]
Fixed it. Thanks for proof reading it.

**Comment by Thinkersbluff** [comment]
> Fixed it. Thanks for proof reading it.

Thank you for writing such complete and readable documentation.  I hope more people take the time to read it before posting issues that are already answered therein.

**Comment by Sebazzz** [comment]
> 
> 
> > Fixed it. Thanks for proof reading it.
> 
> Thank you for writing such complete and readable documentation. I hope more people take the time to read it before posting issues that are already answered therein.

You and me both, mate 😉

**Comment by Thinkersbluff** [comment]
“ Pre-heat PLA and ABS temperatures are properly shown in the touch screen interface”
ABS label was changed to PETG in the stock Creality version.

**Comment by Thinkersbluff** [comment]
"Heaters are turned off when the actual probe is done to decrease electrical noise, which then causes Z-offset and bed leveling inaccuracies"

I find that phrasing to be ambiguous.  
If I may, I suggest this version may be more clear:

"Heaters are turned off while the actual probing is done. This reduces electrical noise which may otherwise cause inaccuracies in the Z-Offset and Bed Leveling measurements."

**Comment by Sebazzz** [comment]
Thanks, updated it

**Comment by Beeble2695** [comment]
Good write-up!
Only point i have resulting from the discord chat there was about terminology:

Support M701/M702 filament load/unload. With @grobux changing it to Extrude/Retract the terminology should be part of the sentence.

**Comment by Thinkersbluff** [comment]
“ If you go to the home/info screen though, you can see it heats ups though.”
Typo. Should read, “ If you go to the home/info screen, though, you can see it heat up.”

**Comment by Thinkersbluff** [comment]
“ After flashing go to the menu and reset to factory settings or issue M502 followed by M500 in Octoprint - do this also if you have been running an earlier release. You need to relevel the bed, reset the z-offset, and reprogram esteps (if you've changed them from defaults).”

I recommend you add “redo PID” to the list of things to be done after flashing, in step 8.  

I use M303 E0 S205 U and M303 E-1 S50 U routinely after M502. The U saves me having to enter the PID parameters, and could be coded into a small .gcode file to print from the SD card if people don’t use their serial port.

**Comment by Thinkersbluff** [comment]
Since G28 now turns off the heaters while probing, and some folks have G28 in their start code, it might help folks to include some advice about reconfiguring their starting gcode to move M109 to after G28 in their start code, if they are currently using Cura and/or M109.  It may even be appreciated if you add a section wth recommended start and stop gcoding with explanations.

**Comment by flofreud** [comment]
>  It may even be appreciated if you add a section wth recommended start and stop gcoding with explanations.

That. I assume that a recommendation of start and stop GCode will reduce the support requests :)

**Comment by Sebazzz** [comment]
> 
> 
> > It may even be appreciated if you add a section wth recommended start and stop gcoding with explanations.
> 
> That. I assume that a recommendation of start and stop GCode will reduce the support requests :)

I'Il add a link [to this page](https://damsteen.nl/blog/2020/10/25/creality-cr6-start-print-without-drooping) - but that G28 cools down after or overrides temperatures is a blocker IMHO, we can't ask people to reslice their stuff.

edit: Opened bug #90 for that.

> Support M701/M702 filament load/unload. With @grobux changing it to Extrude/Retract the terminology should be part of the sentence.

Load/unload is the gcode description - it is unrelated to the touch screen

**Comment by Thinkersbluff** [comment]
> I'Il add a link [to this page](https://damsteen.nl/blog/2020/10/25/creality-cr6-start-print-without-drooping) 

That is a great resource page.  
Hopefully, an FAQ will reduce the probability that a disappointing number of users still post bug reports when they notice that their prints are starting before the nozzle temps get back up to their target settings.

> but that G28 cools down after or overrides temperatures is a blocker IMHO, we can't ask people to reslice their stuff.

Maybe I am different that way - I slice every time.  I don't maintain an inventory of past gcodes that I would not want to "reslice".

**Comment by Sebazzz** [comment]
I've made some additional changes and added some imagery.

**Comment by Botap** [comment]
Hi, I really would like to try the release 5 beta but for some reason I am missing the point on how to download the files. I can't see the 'assets' section :-( Hope you will be able to explain to me. Thanks in advance

**Comment by Thinkersbluff** [comment]
> Hi, I really would like to try the release 5 beta but for some reason I am missing the point on how to download the files. I can't see the 'assets' section :-( Hope you will be able to explain to me. Thanks in advance

These a pre-release notes.  There are no assets yet, because it is not released yet.  If you would like a early look at CF5, you can find a developmental version on their Discord channel. 

https://discord.com/channels/759603746270609428/791082544899555358

**Comment by ThisIsMyUniqueUsername** [comment]
I tried this and it resolved several issues - Z-offset is now saved after reboot, but only when set from the main menu, not the value adjusted during print - could this be also be saved permanently? 

Also I used to set my z-offset to 0.15-0.2, now it's down to 0.1 for correct squish, is this the result of "improved homing accuracy"?


The second issue I noticed is connected to the heater turned off during homing. Because it takes so long my nozzle temp can drop drastically before the print starts, especially in a cold room (it's winter here). Basically the nozzle preheats to 200'C, then homes and by the time it starts extruding the line on the right it's down to about 170'c. With large prints it has enough time to get back to the working temp on skirt, but printing a small item can be problematic. 

Does turning off the heater during homing really improve accuracy that much? I've never had issues with it before, so I'm not sure if the tradeoff is worth it.

Please let me know if you'd like me to log it as a bug or FR separately.

**Comment by Thinkersbluff** [comment]
With Flow Rate and Feed Rate both now having the same speedometer icons, and both being measured in %, it makes the footer display less clear.  Might want to consider a way to clarify that?

**Comment by Sebazzz** [comment]
> 
> 
> With Flow Rate and Feed Rate both now having the same speedometer icons, and both being measured in %, it makes the footer display less clear. Might want to consider a way to clarify that?

This is your cue @grobux 😀

**Comment by BubuMVX** [comment]
Your comment is on point, I'm already thinking about a new design for the footer. There is much more infos to display than before, and the current design isn't suited.

**Comment by BubuMVX** [comment]
> 
> 
> With Flow Rate and Feed Rate both now having the same speedometer icons, and both being measured in %, it makes the footer display less clear. Might want to consider a way to clarify that?

I kept things pretty basic and replaced icons with labels, as I'm lacking space and creativity (for now at least).

The branch `extui-dev` of `CR6Community/CR-6-touchscreen` has the new design.

Raw preview: 

![image](https://user-images.githubusercontent.com/2005901/103250886-79997200-4976-11eb-8d44-d6d80c637924.png)

**Comment by Sebazzz** [comment]
Feel free to go on icon hunt @Thinkersbluff 😉

**Comment by BubuMVX** [comment]
Is the fan speed missing/required on this screen?

**Comment by Sebazzz** [comment]
Fan speed is indeed missing

Met vriendelijke groet,
Sebastiaan Dammann
________________________________
Van: Romain <notifications@github.com>
Verzonden: Tuesday, December 29, 2020 2:07:09 AM
Aan: CR6Community/Marlin <Marlin@noreply.github.com>
CC: Sebastiaan Dammann <sebastiaandammann@outlook.com>; Author <author@noreply.github.com>
Onderwerp: Re: [CR6Community/Marlin] Provisional release notes for CR-6 community firmware release 5 beta (#84)


Is the fan speed missing/required on this screen?

**Comment by Thinkersbluff** [comment]
> Feel free to go on icon hunt @Thinkersbluff 😉

Inspired by Fig 3 in this article: https://link.springer.com/article/10.1007/s40430-020-02610-0

![LoadIcon](https://user-images.githubusercontent.com/36551518/103259186-56ff5d00-4966-11eb-9657-c84e34a580c9.png)

![UNLoadIcon](https://user-images.githubusercontent.com/36551518/103259193-5bc41100-4966-11eb-8711-a9ee9ff33757.png)

If something like this is readable at the desired scale, it may also work for Flow Rate?

**Comment by Sebazzz** [comment]
Let's release this baby.
